### PR TITLE
Improve docker build for local env

### PR DESCRIPTION
**Pull request by ke4**
Changes:
- Change build container to use a gradle image
- parameterise the supporting scripts:
  - execute-single-test
  - debug-single-test

### CODE REVIEW

File: debug-single-test.sh [hunk 1]
```diff
@@ -4,10 +4,56 @@ SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 
 source ${SCRIPT_DIR}/docker/dev.env
 
-TEST_CASE_NAME=$1
-echo "Testing ${TEST_CASE_NAME}"
+function print_usage() {
+  printf '\n%b\n\n' "Usage: ${0} [ -a APPLICATION_NAME ] [ -s SCHEMA_VERSION ]"
+  printf '%b\n' "Debugging a given unit/integration test from the given application using the given schema version"
```
Comment (alfonsomunozpomer) [inline debug-single-test.sh]: I think this wording is simpler and more to the point:
“Debug a unit/integration test in a Gradle sub-project with the given schema version”

Also, I think the usage should include an explanation of the `-n` option.

File: debug-single-test.sh [hunk 2]
```diff
@@ -4,10 +4,56 @@ SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
...
+  printf '\n%b\n' "-p\tName of the project the test can be found\tfor example: app or atlas-web-core"
+  printf '\n%b\n' "-s\tNumeric version of the schema or latest\tfor example: 18 (default is latest)"
```
Comment (alfonsomunozpomer) [inline debug-single-test.sh]:
```suggestion
  printf '\n%b\n' "-p\tName of the sub-project the test can be found;\n\tfor example: app or atlas-web-core (default is app)"
  printf '\n%b\n' "-s\tNumeric version of the schema or latest;\n\tfor example: 18 (default is latest)"
```

File: debug-single-test.sh [hunk 3]
```diff
@@ -4,10 +4,56 @@ SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 
 source ${SCRIPT_DIR}/docker/dev.env
 
-TEST_CASE_NAME=$1
-echo "Testing ${TEST_CASE_NAME}"
+function print_usage() {
+  printf '\n%b\n\n' "Usage: ${0} [ -a APPLICATION_NAME ] [ -s SCHEMA_VERSION ]"
```
Comment (alfonsomunozpomer) [inline debug-single-test.sh]:
```suggestion
  printf '\n%b\n\n' "Usage: ${0} [ -p SUBPROJECT_NAME ] [ -s SCHEMA_VERSION ] -n TEST_NAME"
```

File: execute-single-test.sh [hunk 4]
```diff
@@ -4,10 +4,56 @@ SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 
 source ${SCRIPT_DIR}/docker/dev.env
 
-TEST_CASE_NAME=$1
+function print_usage() {
+  printf '\n%b\n\n' "Usage: ${0} [ -a APPLICATION_NAME ] [ -s SCHEMA_VERSION ]"
+  printf '%b\n' "Executing a given unit/integration test from the given application using the given schema version"
+  printf '\n%b\n' "-p\tName of the project the test can be found\tfor example: app or atlas-web-core"
+  printf '\n%b\n' "-s\tNumeric version of the schema or latest\tfor example: 18 (default is latest)"
+  printf '%b\n\n' "-h\tShow usage instructions"
```
Comment (alfonsomunozpomer) [inline execute-single-test.sh]: Same suggestions as in `debug-single-test.sh` apply here.

**Review by alfonsomunozpomer — CHANGES_REQUESTED** [review]
I suggested some changes in the usage text. Other than that, all good. I don’t know why the first run failed though.

File: debug-single-test.sh (hunk 1 above)
Comment (ke4) [reply debug-single-test.sh]: Good spot. Thanks, fixed both

File: execute-single-test.sh (hunk 4 above)
Comment (ke4) [reply execute-single-test.sh]: Done

File: debug-single-test.sh [hunk 5]
```diff
@@ -4,10 +4,57 @@ SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 
 source ${SCRIPT_DIR}/docker/dev.env
 
-TEST_CASE_NAME=$1
-echo "Testing ${TEST_CASE_NAME}"
+function print_usage() {
+  printf '\n%b\n\n' "Usage: ${0} [ -p SUBPROJECT_NAME ] [ -s SCHEMA_VERSION ] -n TEST_NAME"
+  printf '%b\n' "Debug a unit/integration test in a module with the given schema version"
+  printf '\n%b\n' "-n\tName of the unit/integration test to debug.\tfor example: CellPlotDaoIT"
```
Comment (alfonsomunozpomer) [inline debug-single-test.sh]:
```suggestion
  printf '\n%b\n' "-n\tName of the unit/integration test to debug;\n\tfor example: CellPlotDaoIT"
```

File: execute-single-test.sh [hunk 6]
```diff
@@ -4,10 +4,57 @@ SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 
 source ${SCRIPT_DIR}/docker/dev.env
 
-TEST_CASE_NAME=$1
+function print_usage() {
+  printf '\n%b\n\n' "Usage: ${0} [ -p SUBPROJECT_NAME ] [ -s SCHEMA_VERSION ] -n TEST_NAME"
+  printf '%b\n' "Executing a unit/integration test in a module with the given schema version"
+  printf '\n%b\n' "-n\tName of the unit/integration test to execute.\tfor example: CellPlotDaoIT"
```
Comment (alfonsomunozpomer) [inline execute-single-test.sh]:
```suggestion
  printf '%b\n' "Execute a unit/integration test in a module with the given schema version"
  printf '\n%b\n' "-n\tName of the unit/integration test to execute;\n\tfor example: CellPlotDaoIT"
```

**Review by alfonsomunozpomer — DISMISSED** [review]
Just a couple of typos I could spot (see the suggestions). Other than that, all good.

**Review by alfonsomunozpomer — APPROVED** [review]
:+1:

**Pull request merged**